# FAKE REVIEW CLASSIFICATION

In [141]:
# importing all necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import string
import os
from sklearn import preprocessing, model_selection
from IPython.display import display, HTML
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA

sns.set()

%matplotlib inline

In [142]:
#importing libraries
import numpy as np
import pandas as pd
import nltk
import matplotlib.pyplot as plt
import statistics
import math
import os
from sklearn.model_selection import KFold, StratifiedKFold, cross_val_score
from sklearn.metrics import confusion_matrix
nltk.download('stopwords')
nltk.download('punkt')
nltk.download('wordnet')
nltk.download('omw-1.4')

import re
pd.set_option('max_colwidth', 800)


#importing Libraies
import networkx as nx
import matplotlib.pyplot as plt
from numpy import genfromtxt
import collections
from collections import OrderedDict

import matplotlib.pyplot as plt
%matplotlib inline
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.tree import DecisionTreeClassifier
from sklearn.feature_selection import SelectFromModel
from sklearn.svm import SVC
from sklearn.preprocessing import StandardScaler,MinMaxScaler
from sklearn.linear_model import LinearRegression,LogisticRegression, RidgeClassifier
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis, QuadraticDiscriminantAnalysis
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import classification_report,accuracy_score

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!


In [143]:
#loading dataset
df=pd.read_csv('/content/deceptive.csv')
df.head()

deceptive   hotel  polarity       source  \
0  truthful  conrad  positive  TripAdvisor   
1  truthful   hyatt  positive  TripAdvisor   
2  truthful   hyatt  positive  TripAdvisor   
3  truthful    omni  positive  TripAdvisor   
4  truthful   hyatt  positive  TripAdvisor   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                                                                                                                                                                                                    We stayed for a one night getaway with family on a thursday. Triple AAA rate of 173 was a steal. 7th floor room complete with 44in plasma TV bose stereo, voss and evian water, and gorgeous bathroom(no tub but was fine for us) Concierge was very helpful. You cannot beat this location... Only flaw was breakfast was pricey and service was very very slow(2hours for four kids and four adults on a friday morning) even though there were only two other tables in the restaurant. Food was very good so it was worth the wait. I would return in a heartbeat. A gem in chicago... \n   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Triple A rate with upgrade to view room was less than $200 which also included breakfast vouchers. Had a great view of river, lake, Wrigley Bldg. & Tribune Bldg. Most major restaurants, Shopping, Sightseeing attractions within walking distance. Large room with a very comfortable bed. \n   
2  This comes a little late as I'm finally catching up on my reviews from the past several months:) A dear friend and I stayed at the Hyatt Regency in late October 2007 for one night while visiting a friend and her husband from out of town. This hotel is perfect, IMO. Easy check in and check out. Lovely, clean, comfortable rooms with great views of the city. I know this area pretty well and it's very convenient to many downtown Chicago attractions. We had dinner and went clubing with our friends around Division St.. We had no problems getting cabs back and forth to the Hyatt and there's even public transportation right near by but we didn't bother since we only needed cabs from and to the hotel. Parking, as is usual for Chicago, was expensive but we were able to get our car out quickly (h...   
3                                                                                             The Omni Chicago really delivers on all fronts, from the spaciousness of the rooms to the helpful staff to the prized location on Michigan Avenue. While this address in Chicago requires a high level of quality, the Omni delivers. Check in for myself and a whole group of people with me was under 3 minutes, the staff had plentiful recommendations for dining and events, and the rooms are some of the largest

In [144]:
d = []
for i in range(1600):
    s = i+1
    d.append(s)

In [145]:
#d

In [146]:
df.insert(0, 'Index', d)
df.head()

Index deceptive   hotel  polarity       source  \
0      1  truthful  conrad  positive  TripAdvisor   
1      2  truthful   hyatt  positive  TripAdvisor   
2      3  truthful   hyatt  positive  TripAdvisor   
3      4  truthful    omni  positive  TripAdvisor   
4      5  truthful   hyatt  positive  TripAdvisor   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                                                                                                                                                                                                    We stayed for a one night getaway with family on a thursday. Triple AAA rate of 173 was a steal. 7th floor room complete with 44in plasma TV bose stereo, voss and evian water, and gorgeous bathroom(no tub but was fine for us) Concierge was very helpful. You cannot beat this location... Only flaw was breakfast was pricey and service was very very slow(2hours for four kids and four adults on a friday morning) even though there were only two other tables in the restaurant. Food was very good so it was worth the wait. I would return in a heartbeat. A gem in chicago... \n   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Triple A rate with upgrade to view room was less than $200 which also included breakfast vouchers. Had a great view of river, lake, Wrigley Bldg. & Tribune Bldg. Most major restaurants, Shopping, Sightseeing attractions within walking distance. Large room with a very comfortable bed. \n   
2  This comes a little late as I'm finally catching up on my reviews from the past several months:) A dear friend and I stayed at the Hyatt Regency in late October 2007 for one night while visiting a friend and her husband from out of town. This hotel is perfect, IMO. Easy check in and check out. Lovely, clean, comfortable rooms with great views of the city. I know this area pretty well and it's very convenient to many downtown Chicago attractions. We had dinner and went clubing with our friends around Division St.. We had no problems getting cabs back and forth to the Hyatt and there's even public transportation right near by but we didn't bother since we only needed cabs from and to the hotel. Parking, as is usual for Chicago, was expensive but we were able to get our car out quickly (h...   
3                                                                                             The Omni Chicago really delivers on all fronts, from the spaciousness of the rooms to the helpful staff to the prized location on Michigan Avenue. While this address in Chicago requires a high level of quality, the Omni delivers. Check in for myself and a whole group of people with me was under 3 minutes, the staff had plentiful recommendations for dining and even

In [147]:
df.shape

(1600, 7)

In [148]:
df['Class'].value_counts()

1    800
0    800
Name: Class, dtype: int64

In [149]:
df = df.sample(frac=1).reset_index(drop=True)
df.head()

Index  deceptive     hotel  polarity       source  \
0   1505  deceptive     james  negative        MTurk   
1     76   truthful  fairmont  positive  TripAdvisor   
2   1537  deceptive   allegro  negative        MTurk   
3    687  deceptive    hilton  positive        MTurk   
4   1270  deceptive     hyatt  negative        MTurk   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                               The James Hotel in Chicago was a huge let down. I was expecting a truly spectacular experience. On the website it appears to be an absolutely fabulous state of the art hotel. Well, my family's bad experience started right when we walked in the door. It turns out the room we had reserved had been given to someone else, and the man at the front desk was extremely rude. We had to wait for an hour for the house keeping service to prepare a new room for us. The room had an odd smell and we had to ask for towels and hand soap as they weren't already in the room. The one thing that was good about the place was the fabulous gym. All in all, I feel that we paid way too much for what we got. Next time I will look to stay somewhere else.\n   
1                                                                                                                                          The reviews we read were a bit mixed, but I thought it was excellent. I've stayed in The Splendido, Villa D'Este and others and it doesn't compare, but nor should it. It doesn't claim to be in that space. I expected it to be a notch off the Four Seasons and so it proved. A only a small notch too!. Lovely clean and bright room. Superb views, very quiet, comfy bed etc. Excellent service all round and nice staff. For the money and quality it stacked up very well indeed. Couldn't really fault it any way given it was half the cost of the Four Seasons but pretty close in terms of overall experience. We will go there again when in Chicago. Great location too. \n   
2         I was very unsatisfied with my stay at the Allegro. The front desk clerk seemed very put out when we walked up to her window. After giving her our confirmation number she let us know that our rooms were not ready yet. This came as a surprise to me since we arrived late in the day and check in had already began. After waiting an hour to check in we were not able to get the room features that we had requested. We were informed that they were 'only requests' and not a guarantee. Room service was also very slow and appeared to be understaffed. When we finally got someone to answer the phone they were out of most everything we wanted. The food took an eternity to reach the room and the order wasn't exactly right either. This is not the kind of service I expected from an upscale hotel.\n   
3                                                                                                                                                                                                                                                                                                                                                                                   My stay here was superb. The view from my windows was stunning, as I l

### Data pre-processing

In [150]:
# dropping unnecessary columns. 
df.drop(['deceptive', 'hotel', 'polarity', 'source'], axis = 1, inplace=True)
df.head()

Index  \
0   1505   
1     76   
2   1537   
3    687   
4   1270   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                               The James Hotel in Chicago was a huge let down. I was expecting a truly spectacular experience. On the website it appears to be an absolutely fabulous state of the art hotel. Well, my family's bad experience started right when we walked in the door. It turns out the room we had reserved had been given to someone else, and the man at the front desk was extremely rude. We had to wait for an hour for the house keeping service to prepare a new room for us. The room had an odd smell and we had to ask for towels and hand soap as they weren't already in the room. The one thing that was good about the place was the fabulous gym. All in all, I feel that we paid way too much for what we got. Next time I will look to stay somewhere else.\n   
1                                                                                                                                          The reviews we read were a bit mixed, but I thought it was excellent. I've stayed in The Splendido, Villa D'Este and others and it doesn't compare, but nor should it. It doesn't claim to be in that space. I expected it to be a notch off the Four Seasons and so it proved. A only a small notch too!. Lovely clean and bright room. Superb views, very quiet, comfy bed etc. Excellent service all round and nice staff. For the money and quality it stacked up very well indeed. Couldn't really fault it any way given it was half the cost of the Four Seasons but pretty close in terms of overall experience. We will go there again when in Chicago. Great location too. \n   
2         I was very unsatisfied with my stay at the Allegro. The front desk clerk seemed very put out when we walked up to her window. After giving her our confirmation number she let us know that our rooms were not ready yet. This came as a surprise to me since we arrived late in the day and check in had already began. After waiting an hour to check in we were not able to get the room features that we had requested. We were informed that they were 'only requests' and not a guarantee. Room service was also very slow and appeared to be understaffed. When we finally got someone to answer the phone they were out of most everything we wanted. The food took an eternity to reach the room and the order wasn't exactly right either. This is not the kind of service I expected from an upscale hotel.\n   
3                                                                                                                                                                                                                                                                                                                                                                                   My stay here was superb. The view from my windows was stunning, as I looked out I could see the beautiful Grant Park. The room had a nice airy feel but was also warm and inviting. The bed was very comfortable. The wireless internet was fast and effective which is nice especially when on business trips. THe hotel is in walking distan

In [151]:
# Cleaning Raw tweets
def clean_text(text):
    
    #remove emails
    text = ' '.join([i for i in text.split() if '@' not in i])
    
    #remove web address
    text = re.sub('http[s]?://\S+', '', text)

    #remove brackets and parantheses (and removing the text inside the brackets and parantheses)
    text = re.sub("[\(\[].*?[\)\]]", "", text)

    #remove punctuations
    text = re.sub(r'[^\w\s]', ' ', text)
    
    #Filter to allow only alphabets and numbers
    text = re.sub(r'[^a-zA-Z\']', ' ', text)
    
    #Remove Unicode characters
    text = re.sub(r'[^\x00-\x7F]+', '', text)

    #remove double spaces 
    text = re.sub('\s+', ' ', text)
    
    return text

In [152]:
# applying that into function.
df["clean_text"] = df['text'].apply(lambda x: clean_text(x))
df.head(10)

Index  \
0   1505   
1     76   
2   1537   
3    687   
4   1270   
5   1390   
6   1191   
7    443   
8    879   
9    219   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  \
0                                                               The James Hotel in Chicago was a huge let down. I was expecting a truly spectacular experience. On the website it appears to be an absolutely fabulous state of the art hotel. Well, my family's bad experience started right when we walked in the door. It turns out the room we had reserved had been given to someone else, and the man at the front desk was extremely rude. We had to wait for an hour for the house keeping service to prepare a new room for us. The room had an odd smell and we had to ask for towels and hand soap as they weren't already in the room. The one thing that was good about the place was the fabulous gym. All in all, I feel that we paid way too much for what we got. Next time I will look to stay somewhere else.\n   
1                                                                                                                                          The reviews we read were a bit mixed, but I thought it was excellent. I've stayed in The Splendido, Villa D'Este and others and it doesn't compare, but nor should it. It doesn't claim to be in that space. I expected it to be a notch off the Four Seasons and so it proved. A only a small notch too!. Lovely clean and bright room. Superb views, very quiet, comfy bed etc. Excellent service all round and nice staff. For the money and quality it stacked up very well indeed. Couldn't really fault it any way given it was half the cost of the Four Seasons but pretty close in terms of overall experience. We will go there again when in Chicago. Great location too. \n   
2         I was very unsatisfied with my stay at the Allegro. The front desk clerk seemed very put out when we walked up to her window. After giving her our confirmation number she let us know that our rooms were not ready yet. This came as a surprise to me since we arrived late in the day and check in had already began. After waiting an hour to check in we were not able to get the room features that we had requested. We were informed that they were 'only requests' and not a guarantee. Room service was also very slow and appeared to be understaffed. When we finally got someone to answer the phone they were out of most everything we wanted. The food took an eternity to reach the room and the order wasn't exactly right either. This is not the kind of service I expected from an upscale hotel.\n   
3                                                                                                                                                                                                                                                                                                                                                                                   My stay here was superb. The view from my windows was stunning, as I looked out I could see the beautiful Grant Park. The room had a nice airy feel but was also warm and inviting. The bed was very comfortable. The wireless internet was fast and effective which is nice espec

In [153]:
# dropping the uncleaned text column "text"
df = df.drop(['text'], axis=1)

In [154]:
# renaming the column
df.rename(columns = {'clean_text':'text'}, inplace = True)
df.head()

Index  Class  \
0   1505      0   
1     76      1   
2   1537      0   
3    687      0   
4   1270      0   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              text  
0                                                                             The James Hotel in Chicago was a huge let down I was expecting a truly spectacular experience On the website it appears to be an absolutely fabulous state of the art hotel Well my family s bad experience started right when we walked in the door It turns out the room we had reserved had been given to someone else and the man at the front desk was extremely rude We had to wait for an hour for the house keeping service to prepare a new room for us The room had an odd smell and we had to ask for towels and hand soap as they weren t already in the room The one thing that was good about the place was the fabulous gym All in all I feel that we paid way too much for what we got Next time I will look to stay somewhere else   
1                                                                                                                                                              The reviews we read were a bit mixed but I thought it was excellent I ve stayed in The Splendido Villa D Este and others and it doesn t compare but nor should it It doesn t claim to be in that space I expected it to be a notch off the Four Seasons and so it proved A only a small notch too Lovely clean and bright room Superb views very quiet comfy bed etc Excellent service all round and nice staff For the money and quality it stacked up very well indeed Couldn t really fault it any way given it was half the cost of the Four Seasons but pretty close in terms of overall experience We will go there again when in Chicago Great location too   
2                      I was very unsatisfied with my stay at the Allegro The front desk clerk seemed very put out when we walked up to her window After giving her our confirmation number she let us know that our rooms were not ready yet This came as a surprise to me since we arrived late in the day and check in had already began After waiting an hour to check in we were not able to get the room features that we had requested We were informed that they were only requests and not a guarantee Room service was also very slow and appeared to be understaffed When we finally got someone to answer the phone they were out of most everything we wanted The food took an eternity to reach the room and the order wasn t exactly right either This is not the kind of service I expected from an upscale hotel   
3                                                                                                                                                                                                                                                                                                                                                                                             My stay here was superb The view from my windows was stunning as I looked out I could see the beautiful Grant Park The room had a nice airy feel but was also warm and inviting The bed was very comfortable The wireless internet was fast and effective which is nice especially when on bus

In [155]:
import spacy
#loading the english language small model of spacy
en = spacy.load('en_core_web_sm')
sw_spacy = en.Defaults.stop_words
print(sw_spacy)

{'we', 'regarding', 'sometime', 'go', 'again', 'through', "n't", "'s", '‘d', '‘re', 'herein', 'namely', 'ca', 'too', 'across', 'them', 'often', 'please', 'became', 'call', 'here', 'she', 'both', 'in', 'seems', 'next', 'my', 'hundred', 'will', "'ll", '’s', 'three', 'ever', 'within', 'however', 'others', 'behind', 'therefore', 'wherein', 'one', 'whole', 'nobody', 'else', "'d", 'hence', 'ourselves', 'up', 'quite', 'mine', 'indeed', '‘ve', 'around', 'keep', 'well', 'thereafter', 'since', 'twelve', 'using', 'us', '’d', '‘s', 'latterly', 'n’t', 'any', 'it', 'on', 'was', 'with', 'just', "'m", 'back', 'done', 'beforehand', 'ours', 'first', 'full', 'below', 'yourself', 'perhaps', 'another', 'had', 'this', 'what', 'or', 'hers', 'whence', 'fifty', 'everything', 'as', 'whom', '‘ll', 'upon', 'these', 'themselves', 'me', 'top', 'n‘t', 'none', 'under', 'whereby', 'together', 'out', 'although', 'thereby', 'become', 'your', 'would', 'bottom', 'least', "'re", 'six', 'nevertheless', 'anyhow', 'whatever',

In [156]:
import nltk
from nltk.corpus import stopwords
sw_nltk = stopwords.words('english')
print(sw_nltk)

['i', 'me', 'my', 'myself', 'we', 'our', 'ours', 'ourselves', 'you', "you're", "you've", "you'll", "you'd", 'your', 'yours', 'yourself', 'yourselves', 'he', 'him', 'his', 'himself', 'she', "she's", 'her', 'hers', 'herself', 'it', "it's", 'its', 'itself', 'they', 'them', 'their', 'theirs', 'themselves', 'what', 'which', 'who', 'whom', 'this', 'that', "that'll", 'these', 'those', 'am', 'is', 'are', 'was', 'were', 'be', 'been', 'being', 'have', 'has', 'had', 'having', 'do', 'does', 'did', 'doing', 'a', 'an', 'the', 'and', 'but', 'if', 'or', 'because', 'as', 'until', 'while', 'of', 'at', 'by', 'for', 'with', 'about', 'against', 'between', 'into', 'through', 'during', 'before', 'after', 'above', 'below', 'to', 'from', 'up', 'down', 'in', 'out', 'on', 'off', 'over', 'under', 'again', 'further', 'then', 'once', 'here', 'there', 'when', 'where', 'why', 'how', 'all', 'any', 'both', 'each', 'few', 'more', 'most', 'other', 'some', 'such', 'no', 'nor', 'not', 'only', 'own', 'same', 'so', 'than', '

In [157]:
sw_nltk.extend(['th','stay'])
print(len(sw_nltk))

181


In [158]:
# preprocessing the data (removing stopwords, convert the sentence to lower case, and also applying lemmatization to get meaningful words )
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer, PorterStemmer
#stop_words = set(stopwords.words('english'))
ps=PorterStemmer()
wnl=WordNetLemmatizer()
def clean_lemma(review):
    review = review.lower()
    review = review.split()
    review = [wnl.lemmatize(word) for word in review if word not in sw_nltk]
    review = [wnl.lemmatize(word) for word in review if word not in sw_spacy]
    review = ' '.join(review)
    return review

In [159]:
# using lemmatization
df['text']=df['text'].apply(clean_lemma)
df.head()

Index  Class  \
0   1505      0   
1     76      1   
2   1537      0   
3    687      0   
4   1270      0   

                                                                                                                                                                                                                                                                                                                                                                                                                                   text  
0                                                         james hotel chicago huge let expecting truly spectacular experience website appears absolutely fabulous state art hotel family bad experience started right walked door turn room reserved given man desk extremely rude wait hour house keeping service prepare new room u room odd smell ask towel hand soap room thing good place fabulous gym feel paid way got time look  
1                                                                                       review read bit mixed thought excellent stayed splendido villa este compare claim space expected notch season proved small notch lovely clean bright room superb view quiet comfy bed etc excellent service round nice staff money quality stacked fault way given half cost season pretty close term overall experience chicago great location  
2                                                      unsatisfied allegro desk clerk walked window giving confirmation number let u know room ready came surprise arrived late day check began waiting hour check able room feature requested informed request guarantee room service slow appeared understaffed finally got answer phone wanted food took eternity reach room order exactly right kind service expected upscale hotel  
3                                                                                                                                                                                          superb view window stunning looked beautiful grant park room nice airy feel warm inviting bed comfortable wireless internet fast effective nice especially business trip hotel walking distance museum nice vaca overall thouroughly enjoyed  
4  decided spend little extra hotel visiting son chicago mistake room smaller appear web place dirty stayed best western cleaner air conditioner super loud wall paper thin people room party despite desk security failed stop ruckus thought desk staff rude usually hyatts hiltons etc staff polite try accommodating people particularly personable interruption work reason positive hotel location negative experience save money

In [160]:
df1 = df.sort_values(by=['Index'], ascending=True)
df1.head()

Index  Class  \
1541      1      1   
1434      2      1   
957       3      1   
877       4      1   
1301      5      1   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         text  
1541                                                                                                                                                                                                                                                            stayed night getaway family thursday triple aaa rate steal floor room complete plasma tv bose stereo voss evian water gorgeous bathroom concierge helpful beat location flaw breakfast pricey service slow table restaurant food good worth wait return heartbeat gem chicago  
1434                                                                                                                                                                                                                                                                                                                                 triple rate upgrade view room le included breakfast voucher great view river lake wrigley bldg tribune bldg major restaurant shopping sightseeing attraction walking distance large room comfortable bed  
957   come little late finally catching review past month dear friend stayed hyatt regency late october night visiting friend husband town hotel perfect imo easy check check lovely clean comfortable room great view city know area pretty convenient downtown chicago attraction dinner went clubing friend division st problem getting cab forth hyatt public transportation right near bother needed cab hotel parking usual chicago expensive able car quickly problem best got rate hotwire downright steal area chicago quality hotel  
877                                                                                                                       omni chicago delivers spaciousness room helpful staff prized location michigan avenue address chicago requires high level quality omni delivers check group people minute staff plentiful recommendation dining event room largest find price range chicago standard room separate living area work desk fitness center free weight weight machine row cardio equipment shared room feel cramped way great property  
1301                                                                                                                                                                                                                                                                                                                                asked high floor away elevator got room pleasantly decorated functional clean need lot service pleasant prompt fitness center equipped working order great location end michigan avenue shopping district

In [161]:
df1.reset_index(drop=True, inplace=True)

In [162]:
df1.head()

Index  Class  \
0      1      1   
1      2      1   
2      3      1   
3      4      1   
4      5      1   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      text  
0                                                                                                                                                                                                                                                            stayed night getaway family thursday triple aaa rate steal floor room complete plasma tv bose stereo voss evian water gorgeous bathroom concierge helpful beat location flaw breakfast pricey service slow table restaurant food good worth wait return heartbeat gem chicago  
1                                                                                                                                                                                                                                                                                                                                 triple rate upgrade view room le included breakfast voucher great view river lake wrigley bldg tribune bldg major restaurant shopping sightseeing attraction walking distance large room comfortable bed  
2  come little late finally catching review past month dear friend stayed hyatt regency late october night visiting friend husband town hotel perfect imo easy check check lovely clean comfortable room great view city know area pretty convenient downtown chicago attraction dinner went clubing friend division st problem getting cab forth hyatt public transportation right near bother needed cab hotel parking usual chicago expensive able car quickly problem best got rate hotwire downright steal area chicago quality hotel  
3                                                                                                                      omni chicago delivers spaciousness room helpful staff prized location michigan avenue address chicago requires high level quality omni delivers check group people minute staff plentiful recommendation dining event room largest find price range chicago standard room separate living area work desk fitness center free weight weight machine row cardio equipment shared room feel cramped way great property  
4                                                                                                                                                                                                                                                                                                                                asked high floor away elevator got room pleasantly decorated functional clean need lot service pleasant prompt fitness center equipped working order great location end michigan avenue shopping district

### FEATURE EXTRACTION (TFIDF)

In [163]:
x = df1['text']
y = df1['Class']

In [164]:
y

0       1
1       1
2       1
3       1
4       1
       ..
1595    0
1596    0
1597    0
1598    0
1599    0
Name: Class, Length: 1600, dtype: int64

In [165]:
from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer 
from sklearn.feature_extraction.text import TfidfVectorizer
cv=TfidfVectorizer()
corpii=cv.fit_transform(x) 

In [166]:
feature_names = cv.get_feature_names()
#feature_names

/usr/local/lib/python3.8/dist-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)


In [167]:
len(feature_names)

7905

In [168]:
u1 = pd.DataFrame(corpii.toarray(), columns=feature_names)
u1.head()

aaa  aaahed  aback  abassador  abd  ability      able  abound  abrupt  \
0  0.217034     0.0    0.0        0.0  0.0      0.0  0.000000     0.0     0.0   
1  0.000000     0.0    0.0        0.0  0.0      0.0  0.000000     0.0     0.0   
2  0.000000     0.0    0.0        0.0  0.0      0.0  0.090437     0.0     0.0   
3  0.000000     0.0    0.0        0.0  0.0      0.0  0.000000     0.0     0.0   
4  0.000000     0.0    0.0        0.0  0.0      0.0  0.000000     0.0     0.0   

   abruptly  ...  yucky  yummo  yummy  yunan  yup  zagat  zest  zipped  zone  \
0       0.0  ...    0.0    0.0    0.0    0.0  0.0    0.0   0.0     0.0   0.0   
1       0.0  ...    0.0    0.0    0.0    0.0  0.0    0.0   0.0     0.0   0.0   
2       0.0  ...    0.0    0.0    0.0    0.0  0.0    0.0   0.0     0.0   0.0   
3       0.0  ...    0.0    0.0    0.0    0.0  0.0    0.0   0.0     0.0   0.0   
4       0.0  ...    0.0    0.0    0.0    0.0  0.0    0.0   0.0     0.0   0.0   

   zoo  
0  0.0  
1  0.0  
2  0.0  
3  0.0  
4  0.0  

[5 rows x 7905 columns]

### FEATURE SELECTION (CHI- SQUARE)

In [169]:
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import chi2
# highest chi-squared statistics are selected
chi2_features = SelectKBest(chi2, k = 2000)
X_kbest_features = chi2_features.fit_transform(u1, y)

In [170]:
chi_support = chi2_features.get_support()

In [171]:
chi_feature = u1.loc[:,chi_support].columns.tolist()

In [172]:
X_kbest_features

array([[0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       ...,
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.],
       [0., 0., 0., ..., 0., 0., 0.]])

In [173]:
u3 = pd.DataFrame(X_kbest_features, columns = chi_feature)
u3.head()

absent  absolute  absolutely  absurd  accept  accepted  accesible  \
0     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
1     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
2     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
3     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
4     0.0       0.0         0.0     0.0     0.0       0.0        0.0   

   accomadating  accomidations  accommodated  ...  yellow  yep  yes   yo  \
0           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
1           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
2           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
3           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
4           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   

   yoga  young  youve   yr  yummy  zone  
0   0.0    0.0    0.0  0.0    0.0   0.0  
1   0.0    0.0    0.0  0.0    0.0   0.0  
2   0.0    0.0    0.0  0.0    0.0   0.0  
3   0.0    0.0    0.0  0.0    0.0   0.0  
4   0.0    0.0    0.0  0.0    0.0   0.0  

[5 rows x 2000 columns]

In [174]:
X1 = pd.concat([u3, y], axis=1)
X1.head()

absent  absolute  absolutely  absurd  accept  accepted  accesible  \
0     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
1     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
2     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
3     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
4     0.0       0.0         0.0     0.0     0.0       0.0        0.0   

   accomadating  accomidations  accommodated  ...  yep  yes   yo  yoga  young  \
0           0.0            0.0           0.0  ...  0.0  0.0  0.0   0.0    0.0   
1           0.0            0.0           0.0  ...  0.0  0.0  0.0   0.0    0.0   
2           0.0            0.0           0.0  ...  0.0  0.0  0.0   0.0    0.0   
3           0.0            0.0           0.0  ...  0.0  0.0  0.0   0.0    0.0   
4           0.0            0.0           0.0  ...  0.0  0.0  0.0   0.0    0.0   

   youve   yr  yummy  zone  Class  
0    0.0  0.0    0.0   0.0      1  
1    0.0  0.0    0.0   0.0      1  
2    0.0  0.0    0.0   0.0      1  
3    0.0  0.0    0.0   0.0      1  
4    0.0  0.0    0.0   0.0      1  

[5 rows x 2001 columns]

In [175]:
u3.head()

absent  absolute  absolutely  absurd  accept  accepted  accesible  \
0     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
1     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
2     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
3     0.0       0.0         0.0     0.0     0.0       0.0        0.0   
4     0.0       0.0         0.0     0.0     0.0       0.0        0.0   

   accomadating  accomidations  accommodated  ...  yellow  yep  yes   yo  \
0           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
1           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
2           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
3           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   
4           0.0            0.0           0.0  ...     0.0  0.0  0.0  0.0   

   yoga  young  youve   yr  yummy  zone  
0   0.0    0.0    0.0  0.0    0.0   0.0  
1   0.0    0.0    0.0  0.0    0.0   0.0  
2   0.0    0.0    0.0  0.0    0.0   0.0  
3   0.0    0.0    0.0  0.0    0.0   0.0  
4   0.0    0.0    0.0  0.0    0.0   0.0  

[5 rows x 2000 columns]

In [176]:
df1['Class'].value_counts()

1    800
0    800
Name: Class, dtype: int64

In [177]:
from sklearn.model_selection import train_test_split
from sklearn import tree
from sklearn.metrics import classification_report,accuracy_score
from sklearn import preprocessing
from sklearn.linear_model import LogisticRegression
from sklearn.linear_model import LinearRegression
from sklearn.svm import SVC
from sklearn.svm import LinearSVC
from sklearn import metrics

from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.naive_bayes import MultinomialNB
from sklearn.svm import LinearSVC
from xgboost import XGBClassifier
from xgboost import XGBRegressor
from sklearn.metrics import confusion_matrix, recall_score, precision_score,f1_score
import seaborn as sns

In [178]:
X=u3
Y=y

## Model Building

### CROSS VALIDATION

In [179]:
from sklearn.model_selection import KFold
from sklearn.model_selection import StratifiedKFold
from sklearn.model_selection import cross_val_score
cv = KFold(n_splits=10, random_state=7, shuffle=True)
kf = KFold(n_splits=10)

In [180]:
from sklearn.svm import LinearSVC
models = LinearSVC()
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    models.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(models.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.91125
              precision    recall  f1-score   support

           0       0.91      0.92      0.91       800
           1       0.92      0.91      0.91       800

    accuracy                           0.91      1600
   macro avg       0.91      0.91      0.91      1600
weighted avg       0.91      0.91      0.91      1600

Confusion matrix:  [[734  66]
 [ 76 724]]


In [181]:
from sklearn.linear_model import PassiveAggressiveClassifier
pag=PassiveAggressiveClassifier(max_iter=1000, random_state=7, tol=1e-3)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    pag.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(pag.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.8925
              precision    recall  f1-score   support

           0       0.89      0.90      0.89       800
           1       0.90      0.88      0.89       800

    accuracy                           0.89      1600
   macro avg       0.89      0.89      0.89      1600
weighted avg       0.89      0.89      0.89      1600

Confusion matrix:  [[721  79]
 [ 93 707]]


In [182]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression(max_iter=10000) 
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    lr.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(lr.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.89125
              precision    recall  f1-score   support

           0       0.89      0.89      0.89       800
           1       0.89      0.89      0.89       800

    accuracy                           0.89      1600
   macro avg       0.89      0.89      0.89      1600
weighted avg       0.89      0.89      0.89      1600

Confusion matrix:  [[713  87]
 [ 87 713]]


In [183]:
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier, AdaBoostClassifier
rf = RandomForestClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    rf.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(rf.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.86625
              precision    recall  f1-score   support

           0       0.87      0.86      0.87       800
           1       0.86      0.87      0.87       800

    accuracy                           0.87      1600
   macro avg       0.87      0.87      0.87      1600
weighted avg       0.87      0.87      0.87      1600

Confusion matrix:  [[691 109]
 [105 695]]


In [185]:
gbc = GradientBoostingClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    gbc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(gbc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.829375
              precision    recall  f1-score   support

           0       0.83      0.82      0.83       800
           1       0.83      0.83      0.83       800

    accuracy                           0.83      1600
   macro avg       0.83      0.83      0.83      1600
weighted avg       0.83      0.83      0.83      1600

Confusion matrix:  [[660 140]
 [133 667]]


In [186]:
abc = AdaBoostClassifier(n_estimators=300)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    abc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(abc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.8225
              precision    recall  f1-score   support

           0       0.82      0.82      0.82       800
           1       0.82      0.82      0.82       800

    accuracy                           0.82      1600
   macro avg       0.82      0.82      0.82      1600
weighted avg       0.82      0.82      0.82      1600

Confusion matrix:  [[659 141]
 [143 657]]


In [187]:
import xgboost as xgb
xgbc = xgb.XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, 
              gamma=0,  importance_type='gain',
              interaction_constraints='', learning_rate=0.300000012,
              max_delta_step=0, max_depth=6, min_child_weight=1, 
             n_estimators=200, n_jobs=16,
              num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, subsample=1,
              tree_method='exact', use_label_encoder=False
              )
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    xgbc.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(xgbc.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.839375
              precision    recall  f1-score   support

           0       0.84      0.83      0.84       800
           1       0.83      0.85      0.84       800

    accuracy                           0.84      1600
   macro avg       0.84      0.84      0.84      1600
weighted avg       0.84      0.84      0.84      1600

Confusion matrix:  [[665 135]
 [122 678]]


In [188]:
import lightgbm as lgb
lgb = lgb.LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
               importance_type='split', learning_rate=0.1, max_depth=-1,
               min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
               n_estimators=400, n_jobs=-1, num_leaves=31, objective=None,
               random_state=None, reg_alpha=0.0, reg_lambda=0.0, silent=True,
               subsample=1.0, subsample_for_bin=200000, subsample_freq=0)
folds = StratifiedKFold(n_splits=10, shuffle=True, random_state=7)
y_pred = []
y_exp = []

for train_index, test_index in folds.split(X, Y):
    x_train, x_test = X.iloc[train_index], X.iloc[test_index]
    y_train, y_test = Y[train_index], Y[test_index]
    lgb.fit(x_train, y_train)
    # store result from classification
    y_pred.extend(lgb.predict(x_test))
    # store expected result for this specific fold
    y_exp.extend(y_test)

print("Accuracy: ",  accuracy_score(y_exp, y_pred))
print(classification_report(y_exp, y_pred))
print("Confusion matrix: ", confusion_matrix(y_exp, y_pred))

Accuracy:  0.86
              precision    recall  f1-score   support

           0       0.87      0.85      0.86       800
           1       0.85      0.87      0.86       800

    accuracy                           0.86      1600
   macro avg       0.86      0.86      0.86      1600
weighted avg       0.86      0.86      0.86      1600

Confusion matrix:  [[680 120]
 [104 696]]


### Train test split

In [189]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(u3, y, test_size=0.2, random_state=7, stratify=y)

In [190]:
from sklearn.linear_model import PassiveAggressiveClassifier
pa=PassiveAggressiveClassifier(max_iter=1000, random_state=7, tol=1e-3)
pa.fit(x_train, y_train)
y_pred=pa.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[148  12]
 [ 27 133]]
              precision    recall  f1-score   support

           0       0.85      0.93      0.88       160
           1       0.92      0.83      0.87       160

    accuracy                           0.88       320
   macro avg       0.88      0.88      0.88       320
weighted avg       0.88      0.88      0.88       320

accuracy_score:  0.878125


In [191]:
from sklearn.svm import LinearSVC
lsvm = LinearSVC()
lsvm.fit(x_train, y_train)
y_pred=lsvm.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[145  15]
 [ 16 144]]
              precision    recall  f1-score   support

           0       0.90      0.91      0.90       160
           1       0.91      0.90      0.90       160

    accuracy                           0.90       320
   macro avg       0.90      0.90      0.90       320
weighted avg       0.90      0.90      0.90       320

accuracy_score:  0.903125


In [192]:
rf = RandomForestClassifier(n_estimators=300)
rf.fit(x_train, y_train)
y_pred=rf.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[142  18]
 [ 16 144]]
              precision    recall  f1-score   support

           0       0.90      0.89      0.89       160
           1       0.89      0.90      0.89       160

    accuracy                           0.89       320
   macro avg       0.89      0.89      0.89       320
weighted avg       0.89      0.89      0.89       320

accuracy_score:  0.89375


In [193]:
xgbc = xgb.XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, 
              gamma=0,  importance_type='gain',
              interaction_constraints='', learning_rate=0.300000012,
              max_delta_step=0, max_depth=6, min_child_weight=1, 
             n_estimators=200, n_jobs=16,
              num_parallel_tree=1, random_state=0,
              reg_alpha=0, reg_lambda=1, subsample=1,
              tree_method='exact', use_label_encoder=False
              )
xgbc.fit(x_train, y_train)
y_pred=xgbc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[129  31]
 [ 22 138]]
              precision    recall  f1-score   support

           0       0.85      0.81      0.83       160
           1       0.82      0.86      0.84       160

    accuracy                           0.83       320
   macro avg       0.84      0.83      0.83       320
weighted avg       0.84      0.83      0.83       320

accuracy_score:  0.834375


In [194]:
log = LogisticRegression(max_iter=10000) 
log.fit(x_train, y_train)
y_pred=log.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[142  18]
 [ 14 146]]
              precision    recall  f1-score   support

           0       0.91      0.89      0.90       160
           1       0.89      0.91      0.90       160

    accuracy                           0.90       320
   macro avg       0.90      0.90      0.90       320
weighted avg       0.90      0.90      0.90       320

accuracy_score:  0.9


In [195]:
gbc = GradientBoostingClassifier(n_estimators=300)
gbc.fit(x_train, y_train)
y_pred=gbc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[126  34]
 [ 22 138]]
              precision    recall  f1-score   support

           0       0.85      0.79      0.82       160
           1       0.80      0.86      0.83       160

    accuracy                           0.82       320
   macro avg       0.83      0.82      0.82       320
weighted avg       0.83      0.82      0.82       320

accuracy_score:  0.825


In [196]:
abc = AdaBoostClassifier(n_estimators=300)
abc.fit(x_train, y_train)
y_pred=abc.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[137  23]
 [ 22 138]]
              precision    recall  f1-score   support

           0       0.86      0.86      0.86       160
           1       0.86      0.86      0.86       160

    accuracy                           0.86       320
   macro avg       0.86      0.86      0.86       320
weighted avg       0.86      0.86      0.86       320

accuracy_score:  0.859375


In [197]:
import lightgbm as lgb
lgb = lgb.LGBMClassifier(boosting_type='gbdt', class_weight=None, colsample_bytree=1.0,
               importance_type='split', learning_rate=0.1, max_depth=-1,
               min_child_samples=20, min_child_weight=0.001, min_split_gain=0.0,
               n_estimators=400, n_jobs=-1, num_leaves=31, objective=None,
               random_state=None, reg_alpha=0.0, reg_lambda=0.0, silent=True,
               subsample=1.0, subsample_for_bin=200000, subsample_freq=0)
lgb.fit(x_train, y_train)
y_pred=lgb.predict(x_test)
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
print('accuracy_score: ', accuracy_score(y_test, y_pred))

[[134  26]
 [ 20 140]]
              precision    recall  f1-score   support

           0       0.87      0.84      0.85       160
           1       0.84      0.88      0.86       160

    accuracy                           0.86       320
   macro avg       0.86      0.86      0.86       320
weighted avg       0.86      0.86      0.86       320

accuracy_score:  0.85625


### Using linear SVC with 10-fold cross validation, we achieve 91.1% accuracy.